kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/FilterIT.java

```diff
@@ -278,6 +280,74 @@ void supportsForwardDeferredByAsynchronousBrokerRequest(RequestResponseMarkingFi
+                                                                     ForwardingStyle forwardingStyle)
+            throws ExecutionException, InterruptedException, TimeoutException {
+        var markingFilter = new FilterDefinitionBuilder(RequestResponseMarkingFilterFactory.class.getName())
+                .withConfig("keysToMark", Set.of(LIST_TRANSACTIONS),
+                        "direction", Set.of(direction),
+                        "name", name,
+                        "forwardingStyle", forwardingStyle)
+                .build();
+        try (var tester = mockKafkaKroxyliciousTester((mockBootstrap) -> proxy(mockBootstrap).addToFilters(markingFilter));
+                var kafkaClient = tester.simpleTestClient()) {
+            tester.addMockResponseForApiKey(new ResponsePayload(LIST_TRANSACTIONS, LIST_TRANSACTIONS.latestVersion(), new ListTransactionsResponseData()));
+            ApiVersionsResponseData apiVersionsResponseData = new ApiVersionsResponseData();
+            apiVersionsResponseData.apiKeys()
+                    .add(new ApiVersionsResponseData.ApiVersion().setApiKey(FETCH.id).setMaxVersion(FETCH.latestVersion()).setMinVersion(FETCH.oldestVersion()));
+            tester.addMockResponseForApiKey(new ResponsePayload(API_VERSIONS, API_VERSIONS.latestVersion(), apiVersionsResponseData));
+
+            // In the ASYNCHRONOUS_REQUEST_TO_BROKER case, the filter will send an async list_group
+            // request to the broker and defer the forward of the list transaction response until the list groups
+            // response arrives.
+            if (forwardingStyle == ForwardingStyle.ASYNCHRONOUS_REQUEST_TO_BROKER) {
+                tester.addMockResponseForApiKey(new ResponsePayload(LIST_GROUPS, LIST_GROUPS.latestVersion(), new ListGroupsResponseData()));
+            }
+
+            var responseAFuture = kafkaClient
+                    .get(new Request(LIST_TRANSACTIONS, LIST_TRANSACTIONS.latestVersion(), "client", new ListTransactionsRequestData()));
+
+            var responseBFuture = kafkaClient
+                    .get(new Request(LIST_TRANSACTIONS, LIST_TRANSACTIONS.latestVersion(), "client", new ListTransactionsRequestData()));
+            Response responseA = responseAFuture.get(5, TimeUnit.SECONDS);
+            Response responseB = responseBFuture.get(5, TimeUnit.SECONDS);
+
+            List<Request> listTransactionRequests = tester.getRequestsForApiKey(LIST_TRANSACTIONS);
+            var requestMessageReceivedByBrokerA = listTransactionRequests.get(0).message();
+            String dispatchThreadA = getDispatchThreadName(direction, responseA, requestMessageReceivedByBrokerA);
+            var requestMessageReceivedByBrokerB = listTransactionRequests.get(1).message();
+            String dispatchThreadB = getDispatchThreadName(direction, responseB, requestMessageReceivedByBrokerB);
+            assertThat(dispatchThreadA).isNotEmpty().containsIgnoringCase("eventloop");
```

Comment:
I'm not against this approach, but did you consider having the test filter blow up if it finds it is *not* running on the expected thread.